### PR TITLE
fix(builtin): quickfix entry maker does not used fixed width for file

### PR DESCRIPTION
**Pull request by rudiejd**
# Description

Fixes https://github.com/nvim-telescope/telescope.nvim/issues/2838 and https://github.com/nvim-telescope/telescope.nvim/issues/2121

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [X] Locally tested using hidden and non-hidden files with LSP references / quickfix / definition
image
image

    - Hidden directory: 
image

**Configuration**:
* Neovim version (nvim --version): NVIM v0.10.0-dev-2021+gee2127363 
* Operating system and version: Ubuntu 22.04.3 LTS

# Checklist:

- [X] My code follows the style guidelines of this project (stylua)
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (lua annotations) (N/A)

Apologies if this is a bad solution, I'm still quite new to neovim development. This was something that periodically annoyed me about all of the lists that use the quickfix entry maker 😅 

Questions:
- Is it worth renaming the quickfix entry maker? IMO the name is a bit overloaded now
- Do we need tests for this? From a quick look, it doesn't seem like any of the existing (automated) tests verify display

### CODE REVIEW

File: lua/telescope/make_entry.lua [hunk 1]
```diff
@@ -452,37 +452,25 @@ function make_entry.gen_from_quickfix(opts)
...
 
+
     if show_line then
       local text = entry.text
       if opts.trim_text then
         text = text:gsub("^%s*(.-)%s*$", "%1")
       end
       text = text:gsub(".* | ", "")
-      table.insert(input, text)
+      display_string = display_string .. " " .. text
```
Comment (jamestrew) [inline lua/telescope/make_entry.lua:470]:
```suggestion
      display_string = display_string .. ":" .. text
```

Maybe to stay consistent with other entry makers (vimgrep)

File: lua/telescope/make_entry.lua [hunk 2]
```diff
@@ -452,37 +452,25 @@ function make_entry.gen_from_quickfix(opts)
...
+    local display_string = string.format("%s:%d:%d", display_filename, entry.lnum, entry.col)
+    if hidden then
+      display_string = string.format("%4d:%2d", entry.lnum, entry.col)
     end
 
+
     if show_line then
       local text = entry.text
       if opts.trim_text then
         text = text:gsub("^%s*(.-)%s*$", "%1")
```
Comment (jamestrew) [inline lua/telescope/make_entry.lua:467]:
```suggestion
        text = vim.trim(text)
```
Not your change but little refactor

File: lua/telescope/make_entry.lua [hunk 3]
```diff
@@ -452,37 +452,25 @@ function make_entry.gen_from_quickfix(opts)
   local show_line = vim.F.if_nil(opts.show_line, true)
 
   local hidden = utils.is_path_hidden(opts)
-  local items = {
-    { width = vim.F.if_nil(opts.fname_width, 30) },
```
Comment (jamestrew) [inline lua/telescope/make_entry.lua:456]: This means no more `fname_width` option for anything that uses this entry maker right?

Probably should remove that option from the docs (docstring in `builtin/init.lua`).

**Review by jamestrew — CHANGES_REQUESTED** [review]
Looks good so far.

**Comment by jamestrew** [comment]
I think there's a couple more missing
https://github.com/nvim-telescope/telescope.nvim/pull/2842/files#diff-1550bb3ef180a28235d51137ee3e2f1f34f115f272ec4b1fad9ea528f0bf9713L431-L439

**Comment by rudiejd** [comment]
oops right you are, sorry I thought I got those 😅

**Review by jamestrew — APPROVED** [review]
lgtm thanks :+1:

**Comment by jamestrew** [comment]
@Conni2461 This is a minor breaking change, altering the behavior of `gen_from_quickfix` and removing the `fname_width` option.
Planning to merge this with a `!` in the commit message (undecided on feat vs fix). Also wondering if this calls for an addition to the changelog and new comment in the breaking change issue. Want to run this by you before doing so to confirm.

**Comment by Conni2461** [comment]
thanks for your work here :D to both of you. Dont think we need a changelog addition. i think most will be happy with the changes here :laughing:

**Comment by painhardcore** [comment]
@jamestrew @Conni2461 it's sad that it was not included in v0.1.16. Will it happen in v0.1.17?

**Comment by jamestrew** [comment]
@painhardcore This change was considering a breaking change so it'll be 0.2.0.

**Pull request merged**